pkg/controller/common_test.go

```diff
@@ -451,7 +452,7 @@ func convertToNode(obj client.Object) *v1.Node {
 }
 
 func initState(client client.Client, azClient azdisk.Interface, kubeClient kubernetes.Interface, objs ...runtime.Object) (c *SharedState) {
-	c = NewSharedState(consts.DefaultDriverName, consts.DefaultAzureDiskCrdNamespace, consts.WellKnownTopologyKey, &record.FakeRecorder{}, client, azClient, kubeClient, nil)
+	c = NewSharedState(consts.DefaultDriverName, consts.DefaultAzureDiskCrdNamespace, consts.WellKnownTopologyKey, &record.FakeRecorder{}, client, azClient, &provisioner.CrdProvisioner{}, kubeClient, nil)
```

Comment:
Instead of using an uninitialized `provisioner.CrdProvisioner` here, consider adding a `FakeDriverLifecycle` or using [`gomock`](https://github.com/golang/mock) to create a mock.